### PR TITLE
Established Routes and Tweaked Code

### PR DESCRIPTION
**Pull request by javidangarcia**
- Added routes for searching so that users can navigate to stock pages directly.
- Grouped stock endpoints into a router for future use.
- Used Promise.all to await multiple GET requests simultaneously.

Screenshot 2023-07-05 at 6 27 35 PM

### CODE REVIEW

File: backend/routes/stock.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import express from "express";
```
Comment (BriptimusPrimus) [inline backend/routes/stock.js:1]: 👍

File: backend/server.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import express from "express";
-import { sequelize } from "./database.js";
-import { Stock } from "./models/stock.js";
 import cors from "cors";
 import morgan from "morgan";
+import { sequelize } from "./database.js";
```
Comment (BriptimusPrimus) [inline backend/server.js:4]: It seems like sequelize is not needed in this file anymore

File: frontend/src/components/App/App.jsx [hunk 3]
```diff
@@ -1,26 +1,22 @@
 import "./App.css";
 import Navbar from "../Navbar/Navbar";
+import Home from "../Home/Home";
 import Search from "../Search/Search";
-import { useState } from "react";
+import StockData from "../StockData/StockData";
+import { BrowserRouter, Routes, Route } from "react-router-dom";
 
 export default function App() {
-    const [searchInput, setSearchInput] = useState("");
```
Comment (BriptimusPrimus) [inline frontend/src/components/App/App.jsx:7]: 👍

File: frontend/src/components/StockData/StockData.jsx [hunk 4]
```diff
@@ -62,11 +74,14 @@ export default function StockData({ currentStock }) {
...
-                    <p>Sector: {stockData.sector}</p>
-                    <p>Price: {stockData.price}</p>
+                    <div>
+                        <p>Ticker: {stockData.ticker}</p>
+                        <p>Name: {stockData.name}</p>
+                        <p>Description: {stockData.description}</p>
+                        <p>Sector: {stockData.sector}</p>
+                        <p>Price: ${stockData.price}</p>
+                    </div>
+                    <img src={stockData.logo} alt={`This is a logo of ${stockData.name}.`} />
```
Comment (BriptimusPrimus) [inline frontend/src/components/StockData/StockData.jsx:84]: It looks so good with an image. Good job.

File: backend/server.js (hunk 2 above)
Comment (javidangarcia) [reply backend/server.js:4]: At the very bottom of server.js there is a line of code that uses sequelize. I'm pretty sure it updates the database when things change.

File: frontend/src/components/StockData/StockData.jsx [hunk 5]
```diff
@@ -13,14 +15,14 @@ export default function StockData({ currentStock }) {
...
+                const databaseResponse = await axios.get(`http://localhost:3000/stocks/${ticker}`);
                 const stockExistsInDatabase = databaseResponse.data ? true : false;
 
                 if (stockExistsInDatabase) {
                     setStockData(databaseResponse.data);
                 } else {
-                    const stockOverviewUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${currentStock}&apikey=${ALPHA_API_KEY}`;
-                    const stockPriceUrl = `https://finnhub.io/api/v1/quote?symbol=${currentStock}&token=${FINNHUB_API_KEY}`;
+                    const stockOverviewUrl = `https://www.alphavantage.co/query?function=OVERVIEW&symbol=${ticker}&apikey=${ALPHA_API_KEY}`;
+                    const stockPriceUrl = `https://finnhub.io/api/v1/quote?symbol=${ticker}&token=${FINNHUB_API_KEY}`;
```
Comment (vandanasub) [inline frontend/src/components/StockData/StockData.jsx]: I recommend creating constants for URLs, like you've done with API keys

Comment (vandanasub) [inline frontend/src/components/StockData/StockData.jsx]: You can look into `URL` API, which provides builders, making it cleaner and simpler to add on various search parameters to a base URL

**Pull request merged**